### ISSUE TITLE
NameError while using custom mailer in devise

### ISSUE THREAD

**Issue by a-ta-ta** [mperham/sidekiq#3942]
Ruby version: 2.4.1p111
Rails version: 5.1
Sidekiq version: 5.1.3

I'm using sidekiq as adapter for sending devise mails. With the default devise mailer everything was perfect. But I'd like to replace default devise mails.

I followed the instructions from the devise wiki ([https://github.com/plataformatec/devise/wiki/How-To:-Use-custom-mailer](https://github.com/plataformatec/devise/wiki/How-To:-Use-custom-mailer)) and everything is going well in development (where Sidekiq is off), but I do not receive mails in production and from sidekiq control panel i get NameError: uninitialized constant MyMailer.

I checked through console that MyMailer is loaded in Rails, but seems to be 'invisible' for Sidekiq...

I've read quite a lot here (and on SO too) about the same problems happening in development due to eager_load set to false, but I face it in production where eager_load is on (I checked that).

Any help is appreciated!

**Comment by a-ta-ta** [comment]
I added initializer with:
Sidekiq::Extensions.enable_delay!

This time I got Uninitialized constant Sidekiq::Extensions::DelayedMailer

**Comment by mperham** [comment]
Give us an app that reproduces the problem and we'll tell you what's wrong.  There's too many moving parts to debug otherwise.

**Comment by a-ta-ta** [comment]
Recently, I figured out that sidekiq docs have changed. When you have everything installedand working, you normally do not return to 'Basics' or 'Get started' sections. 
But hopless to find any solution, I was clicking everything and found that 'devise-async' gem is no longer the best decision to use.
So I just removed 'devise-async' and set up a worker as was described in the docs.
Everything works! Better than before!
